coremltools/converters/mil/frontend/torch/ops.py

```diff
@@ -4383,7 +4384,7 @@ def to(context, node):
         np_type = nptype_from_builtin(target_dtype.dtype)
         dtype = NUMPY_DTYPE_TO_TORCH_NUM[np_type]
 
-    torch_dtype = NUM_TO_TORCH_DTYPE[dtype]
+    torch_dtype = dtype_to_32bit(dtype)
```

Comment:
The usage here should be something like
```
torch_dtype = dtype_to_32bit(NUM_TO_TORCH_DTYPE[dtype])
```
Concretely, 2 concepts are involved here:
1. Historically, `torch.ji.trace` uses numbers to represent dtypes, so in coremltools we have `NUM_TO_TORCH_DTYPE` to map those torch script numbers back to torch types
2. Core ML does not support 64-bit dtypes, so we need to further cast 64-bit torch dtypes to 32-bit counterpart

@jakesabathia2 do you think this double wrap the best way to go?

Reply:
I have made the changes to how its being used (thanks for providing that background also 🤙 ).
The use cases in the quantization.py seems to be different, should be it altered (like so)?
```   
torch_dtype = dtype_to_32bit(NUM_TO_TORCH_DTYPE.get(torch_dtype_var.val))
```

Reply:
> torch_dtype = dtype_to_32bit(NUM_TO_TORCH_DTYPE.get(torch_dtype_var.val))

Yes, this is the correct change 👍 and makes the dtype translation much clearer than what we had

> The use cases in the quantization.py seems to be different, should be it altered (like so)?

I took a look at other occurrence of `NUM_TO_TORCH_DTYPE`. Looks like only `ops.py` needs this 64-bit to 32-bit replacement: e.g. in quantization we mainly care about pytorch qdtypes, so yeah no need to replace 64 with 32